Doesn't support dict_get with a tuple of column

Cannot run `SELECT dict_get('taxi_zone_dictionary', ('Borough','Zone'), 132)`

> Caused by: java.sql.SQLException: Unknown data type: Borough
        at com.timeplus.misc.Validate.isTrue(Validate.java:41)
        at com.timeplus.data.DataTypeFactory.get(DataTypeFactory.java:100)
        at com.timeplus.data.type.complex.DataTypeTuple.lambda$static$0(DataTypeTuple.java:39)
        at com.timeplus.data.DataTypeFactory.get(DataTypeFactory.java:73)
        at com.timeplus.data.DataTypeFactory.get(DataTypeFactory.java:60)
        at com.timeplus.data.Block.readFrom(Block.java:46)
        at com.timeplus.protocol.DataResponse.readFrom(DataResponse.java:32)
        at com.timeplus.protocol.Response.readFrom(Response.java:33)
        at com.timeplus.client.NativeClient.receiveResponse(NativeClient.java:207)
        at com.timeplus.client.NativeClient.lambda$receiveQuery$0(NativeClient.java:164)
        at com.timeplus.stream.TimeplusQueryResult.consumeDataResponse(TimeplusQueryResult.java:102)
        at com.timeplus.stream.TimeplusQueryResult.ensureHeaderConsumed(TimeplusQueryResult.java:93)
        at com.timeplus.stream.TimeplusQueryResult.header(TimeplusQueryResult.java:54)
        at com.timeplus.jdbc.statement.TimeplusStatement.lambda$executeUpdate$0(TimeplusStatement.java:102)
        at com.timeplus.misc.ExceptionUtil.rethrowSQLException(ExceptionUtil.java:76)
        ... 4 more

Running `SELECT dict_get('taxi_zone_dictionary', 'Borough', 132)` is okay

## Setup
```sql
CREATE DICTIONARY taxi_zone_dictionary
(
  `LocationID` uint16 DEFAULT 0,
  `Borough` string,
  `Zone` string,
  `service_zone` string
)
PRIMARY KEY LocationID
SOURCE(HTTP(URL 'https://datasets-documentation.s3.eu-west-3.amazonaws.com/nyc-taxi/taxi_zone_lookup.csv' FORMAT 'CSVWithNames'))
LIFETIME(MIN 0 MAX 0)
LAYOUT(HASHED_ARRAY());
````

Test code
```java
/*
 * This Java source file was generated by the Gradle 'init' task.
 */
package test_jdbc_driver;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class App {

    public static void main(String[] args) throws Exception {
        String url =
            "jdbc:timeplus://localhost:8463?user=admin&password=changeme";
        try (Connection connection = DriverManager.getConnection(url)) {
            try (Statement stmt = connection.createStatement()) {
                testDictGetSingle(stmt);
                testDictGetTuple(stmt);
            }
        }
    }

    static void testDictGetSingle(Statement stmt) throws SQLException {
        stmt.executeQuery(
            "SELECT dict_get('taxi_zone_dictionary', 'Borough', 132)"
        );
    }

    static void testDictGetTuple(Statement stmt) throws SQLException {
        stmt.executeQuery(
            "SELECT dict_get('taxi_zone_dictionary', ('Borough','Zone'), 132)"
        );
    }
}

```

--- a/timeplus-native-jdbc/src/main/java/com/timeplus/data/type/complex/DataTypeTuple.java
+++ b/timeplus-native-jdbc/src/main/java/com/timeplus/data/type/complex/DataTypeTuple.java
@@ -35,8 +35,23 @@ public class DataTypeTuple implements IDataType<TimeplusStruct, Struct> {
         Validate.isTrue(lexer.character() == '(');
         List<IDataType<?, ?>> nestedDataTypes = new ArrayList<>();
 
+        Integer count = 0;
+        List<String> tupleDataNames = new ArrayList<>();
         for (; ; ) {
-            nestedDataTypes.add(DataTypeFactory.get(lexer, serverContext));
+            while (count < 2) {
+                String tmpName = String.valueOf(lexer.tryGetBareWord());
+                try {
+                    nestedDataTypes.add(DataTypeFactory.get(lexer, serverContext));
+                    count = 0;
+                    break;
+                } catch (Exception e) {
+                    count++;
+                    tupleDataNames.add(tmpName);
+                    if (count >= 2) {
+                        throw e;
+                    }
+                }
+            }
             char delimiter = lexer.character();
             Validate.isTrue(delimiter == ',' || delimiter == ')');
             if (delimiter == ')') {
@@ -44,6 +59,10 @@ public class DataTypeTuple implements IDataType<TimeplusStruct, Struct> {
                 for (int i = 0; i < nestedDataTypes.size(); i++) {
                     if (i > 0)
                         builder.append(",");
+                    if (tupleDataNames.size() > 0) {
+                        builder.append(tupleDataNames.get(i));
+                        builder.append(" ");
+                    }
                     builder.append(nestedDataTypes.get(i).name());
                 }
                 return new DataTypeTuple(builder.append(")").toString(), nestedDataTypes.toArray(new IDataType[0]));

--- a/timeplus-native-jdbc/src/main/java/com/timeplus/data/type/complex/DataTypeTuple.java
+++ b/timeplus-native-jdbc/src/main/java/com/timeplus/data/type/complex/DataTypeTuple.java
@@ -38,15 +38,20 @@ public class DataTypeTuple implements IDataType<TimeplusStruct, Struct> {
         Integer count = 0;
         List<String> tupleDataNames = new ArrayList<>();
         for (; ; ) {
+            // Tuple return types can be specified as either "name + type" or just "type."
+            // Example: tuple(a string, b string) or tuple(string, string)
+            // We use the parameter count to determine the possible return types.
+            // If the first word is not a type, it's treated as a tuple name.
+            // The next word must be a data type; otherwise, an error is reported.
             while (count < 2) {
-                String tmpName = String.valueOf(lexer.tryGetBareWord());
+                String elemName = String.valueOf(lexer.bareWordView());
                 try {
                     nestedDataTypes.add(DataTypeFactory.get(lexer, serverContext));
                     count = 0;
                     break;
                 } catch (Exception e) {
                     count++;
-                    tupleDataNames.add(tmpName);
+                    tupleDataNames.add(elemName);
                     if (count >= 2) {
                         throw e;
                     }
@@ -59,7 +64,7 @@ public class DataTypeTuple implements IDataType<TimeplusStruct, Struct> {
                 for (int i = 0; i < nestedDataTypes.size(); i++) {
                     if (i > 0)
                         builder.append(",");
-                    if (tupleDataNames.size() > 0) {
+                    if (i < tupleDataNames.size()) {
                         builder.append(tupleDataNames.get(i));
                         builder.append(" ");
                     }

--- a/timeplus-native-jdbc/src/main/java/com/timeplus/misc/SQLLexer.java
+++ b/timeplus-native-jdbc/src/main/java/com/timeplus/misc/SQLLexer.java
@@ -166,6 +166,34 @@ public class SQLLexer {
         throw new SQLException("Expect Bare Token.");
     }
 
+    public StringView tryGetBareWord() throws SQLException {
+        skipAnyWhitespace();
+        // @formatter:off
+        if (isCharacter('`')) {
+            return stringLiteralWithQuoted('`');
+        } else if (isCharacter('"')) {
+            return stringLiteralWithQuoted('"');
+        } else if (data.charAt(pos) == '_'
+               || (data.charAt(pos) >= 'a' && data.charAt(pos) <= 'z')
+               || (data.charAt(pos) >= 'A' && data.charAt(pos) <= 'Z')) {
+            int start = pos;
+            for (pos++; pos < data.length(); pos++) {
+                if (!('_' == data.charAt(pos)
+                  || (data.charAt(pos) >= 'a' && data.charAt(pos) <= 'z')
+                  || (data.charAt(pos) >= 'A' && data.charAt(pos) <= 'Z')
+                  || (data.charAt(pos) >= '0' && data.charAt(pos) <= '9'))) {
+                    break;
+                }
+            }
+            // the pos won't change
+            int end = pos;
+            pos = start;
+            return new StringView(data, start, end);
+        }
+        // @formatter:on
+        throw new SQLException("Expect Bare Token.");
+    }
+
     public boolean isWhitespace() {
         return data.charAt(pos++) == ' ';
     }

--- a/timeplus-native-jdbc/src/main/java/com/timeplus/misc/SQLLexer.java
+++ b/timeplus-native-jdbc/src/main/java/com/timeplus/misc/SQLLexer.java
@@ -166,7 +166,8 @@ public class SQLLexer {
         throw new SQLException("Expect Bare Token.");
     }
 
-    public StringView tryGetBareWord() throws SQLException {
+    // Diferrence from bareWord(): the pos won't change after get the word
+    public StringView bareWordView() throws SQLException {
         skipAnyWhitespace();
         // @formatter:off
         if (isCharacter('`')) {
@@ -185,7 +186,6 @@ public class SQLLexer {
                     break;
                 }
             }
-            // the pos won't change
             int end = pos;
             pos = start;
             return new StringView(data, start, end);

--- a/timeplus-native-jdbc/src/test/java/com/timeplus/jdbc/type/TupleTypeTest.java
+++ b/timeplus-native-jdbc/src/test/java/com/timeplus/jdbc/type/TupleTypeTest.java
@@ -24,6 +24,7 @@ import java.math.BigInteger;
 import java.math.RoundingMode;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
 import java.time.ZonedDateTime;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
@@ -200,6 +201,25 @@ public class TupleTypeTest extends AbstractITest implements BytesHelper {
             }
             assertEquals(size, rowCnt);
             statement.execute("DROP STREAM IF EXISTS tuple_test");
+        });
+    }
+
+    @Test
+    public void testTupleWithName() throws Exception {
+        withStatement(statement -> {
+            statement.execute("DROP DICTIONARY IF EXISTS tuple_test");
+            statement.execute("CREATE DICTIONARY tuple_test (LocationID uint16 DEFAULT 0,"
+                                +"Borough string, Zone string, service_zone string) PRIMARY KEY LocationID\n"
+                                +"SOURCE(HTTP(URL 'https://datasets-documentation.s3.eu-west-3.amazonaws.com/nyc-taxi/taxi_zone_lookup.csv'"
+                                +"FORMAT 'CSVWithNames')) \n LIFETIME(MIN 0 MAX 0) LAYOUT(HASHED_ARRAY());");
+
+            ResultSet rs = statement.executeQuery("SELECT dict_get('tuple_test', ('Borough','Zone'), 132)");
+            while (rs.next()) {
+                ResultSetMetaData data = rs.getMetaData();
+                String name = data.getColumnTypeName(1);
+                assertEquals("tuple(Borough string,Zone string)", name);
+            }
+            statement.execute("DROP DICTIONARY IF EXISTS tuple_test");
         });
     }
 

